mlir-tensorrt/tensorrt/lib/TensorRT/IR/Verification.cpp

```diff
@@ -955,12 +955,11 @@ LogicalResult tensorrt::ResizeNearestOp::verify() {
   }
 
   for (int64_t i = 0; i < resizeDim; ++i) {
-    if (inputType.isDynamicDim(outputRank - 1 - i) ||
-        outputType.isDynamicDim(outputRank - 1 - i)) {
-      if (!getScales().has_value())
+    if (inputType.isDynamicDim(outputRank - 1 - i)) {
+      if (!getOutputShape())
         return emitOpError(
-            "output innermost min(3, rank(input)) dimension that resize on "
-            "cannot be dynamic when resize scales parameter is NOT "
```

Comment:
@christopherbate I wonder if this is too strict, for example even with input spatial dims are dynamic, if `scales` is given, the op can still be valid, because I saw this function infers the output shape: https://github.com/NVIDIA/TensorRT-Incubator/blob/main/mlir-tensorrt/tensorrt/lib/TensorRT/IR/TypeInferenceInterfaceImpls.cpp#L1219

Reply:
If `output_shape` is given, you can skip the entire outer `for` loop since.

If `output_shape` is not given, then you must have either 

1. The output result shape is static
2. All input dimensions are static and  the resize scales are given.

Basically you just need to return an error if the output shape cannot somehow be inferred (statically or dynamically).

Reply:
Updated, thanks